Source/GSXib5KeyedUnarchiver.m

```diff
@@ -0,0 +1,3391 @@
+- (id) initForReadingWithData: (NSData*)data
+{
+  // Check data...
+  if (data == nil)
+    {
+      DESTROY(self);
+      return nil;
+    }
+#if     GNUSTEP_BASE_HAVE_LIBXML
+  else
+    {
+      // Ensure we have a XIB 5 version...first see if we can parse the XML...
+      NSXMLDocument *document = [[NSXMLDocument alloc] initWithData: data
+                                                            options: 0
+                                                              error: NULL];
+      if (document == nil)
+        {
+          NSLog(@"%s:DOCUMENT IS NIL: %@\n", __PRETTY_FUNCTION__, document);
+          DESTROY(self);
+          return nil;
+        }
+      else
+        {
+          // Test to see if this is an Xcode 5 XIB...
+          NSArray *documentNodes = [document nodesForXPath: @"/document" error: NULL];
+
+          // Need at LEAST ONE document node...we should find something a bit more
+          // specific to check here...
+          if ([documentNodes count] == 0)
+            {
+              NSWarnMLog(@"XIB is NOT a version 5 level");
+              DESTROY(self);
+            }
+          else
+            {
+              NSXMLParser *theParser  = nil;
+
+              // Initialize...
+              [self _initCommon];
+
+              // Createe the parser and parse the data...
+              theParser = [[NSXMLParser alloc] initWithData: data];
+              [theParser setDelegate: self];
+
+              NS_DURING
+                {
+                  // Parse the XML data
+                  [theParser parse];
+                }
+              NS_HANDLER
+                {
+                  NSLog(@"Exception occurred while parsing Xib: %@", [localException reason]);
+                  DESTROY(self);
+                }
+              NS_ENDHANDLER
+
+              DESTROY(theParser);
+
+              // Decode optional resources
+              _resources = RETAIN([self decodeObjectForKey: @"resources"]);
+            }
+        }
+    }
+#else
+  else
+    {
+      DESTROY(self);
+    }
+#endif
+
+  return self;
+}
+
+- (void) _initCommon
+{
+  [super _initCommon];
+
+  _orderedObjectsDict = RETAIN([NSMutableDictionary dictionary]);
+
+  // Create our object(s)...
+  _orderedObjects      = [[GSXibElement alloc] initWithType: @"array"
+                                              andAttributes: [NSDictionary dictionaryWithObjectsAndKeys:
+                                                                             @"orderedObjects", @"key",
+                                                                           nil]];
+  _objectRecords       = [[GSXibElement alloc] initWithType: @"object"
+                                              andAttributes: [NSDictionary dictionaryWithObjectsAndKeys:
+                                                                             @"IBMutableOrderedSet", @"class",
+                                                                           @"objectRecords", @"key",
```

Comment:
This indentation for arguments is not even consistent with itself

Reply:
The indentation is the Emacs default indentation for Objective-C. I agree it looks wiered

Reply:
While 'whatever emacs does' would be a good idea, I don't think Emacs has done a stellar job here or that it's an official policy? :/